track_generator/ground_truth_generator.py

```diff
@@ -0,0 +1,67 @@
+from xml.etree import cElementTree as ET, ElementTree
+from xml.etree import ElementTree
+from xml.dom import minidom
+
+from track_generator.track import Track, Start, Straight, Arc, Crosswalk, Intersection, Gap, ParkingArea, TrafficIsland, Clothoid
+
+class GroundTruthGenerator:
+
+    def __init__(self, track_name: str, output_directory: str):
+        self.track_name = track_name
+        self.output_directory = output_directory
+        self.root = ET.Element('TrackDefinition', {'version': '0.0.1'})
+        self.points = ET.SubElement(self.root, 'Points')
+
+    def generate_ground_truth(self, track: Track):
+        for segment in track.segments:
+            self.generate_segment(segment)
+        with open(self.output_directory + 'ground_truth.xml', 'w') as f:
+            f.write(minidom.parseString(ET.tostring(self.root, 'utf-8')).toprettyxml(indent='\t'))
+
+    def generate_segment(self, segment):
+        if isinstance(segment, Start):
+            pass
+        elif isinstance(segment, (Straight, ParkingArea, Gap, Crosswalk)):
+            self.generate_straight(segment)
+        elif isinstance(segment, Arc):
+            self.generate_arc(segment)
+        elif isinstance(segment, Intersection):
+            self.generate_intersection(segment)
+        elif isinstance(segment, TrafficIsland):
+            self.generate_traffic_island(segment)
+        elif isinstance(segment, Clothoid):
+            self.generate_clothoid(segment)
+        else:
+            raise RuntimeError()
+        
+    def generate_straight(self, segment: Straight):
+        for i,_ in enumerate(segment.center_line_polygon):
```

Comment:
Warum benutzt du hier noch die "normalen" python iterator for loops? Also `for point in segment.center_line_polygon:` ?

Reply:
Weil mehrere Listen iteriert werden. (siehe Zeile 39 und 40)